client/state/terms/actions.js

```diff
@@ -1,15 +1,77 @@
 /**
+ * External dependencies
+ */
+import omit from 'lodash/omit';
+
+/**
  * Internal dependencies
  */
 import wpcom from 'lib/wp';
 import {
+	TERMS_ADD_REQUEST,
+	TERMS_ADD_REQUEST_SUCCESS,
+	TERMS_ADD_REQUEST_FAILURE,
 	TERMS_RECEIVE,
 	TERMS_REQUEST,
 	TERMS_REQUEST_SUCCESS,
 	TERMS_REQUEST_FAILURE
 } from 'state/action-types';
 
 /**
+ * Module constants
+ */
+let temporaryIdCount = 0;
```

Comment:
My first reaction is that module-level variables are a code smell in actions.js, but thinking on this further, it's not much unlike [our existing notice actions](https://github.com/Automattic/wp-calypso/blob/master/client/state/notices/actions.js). As background, there's some back and forth in #1496 on whether we could avoid this technique. It seems unless we want to put this responsibility on the component dispatching calling the action creator, we'd want to be content with a bit of magic.

Two suggestions I might make though:

- We should just use `lodash/uniqueId`, optionally with a prefix ([docs](https://lodash.com/docs#uniqueId))
- Once we use `uniqueId` it probably doesn't warrant its own utility function. Just inline in `addTerm`, e.g. `const temporaryId = uniqueId( 'term' );`